src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -841,14 +787,106 @@ bool SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets, u
   return true;
 }
 
+std::unique_ptr<Uptane::Target> SotaUptaneClient::findTargetHelper(const Uptane::Targets &cur_targets,
+                                                                   const Uptane::Target &queried_target, int level,
+                                                                   bool terminating) {
+  auto it = std::find(cur_targets.targets.cbegin(), cur_targets.targets.cend(), queried_target);
+  if (it != cur_targets.targets.cend()) {
+    return std_::make_unique<Uptane::Target>(*it);
+  }
+
+  if (terminating || level >= Uptane::kDelegationsMaxDepth) {
+    return std::unique_ptr<Uptane::Target>(nullptr);
+  }
+
+  for (const auto &delegate_name : cur_targets.delegated_role_names_) {
+    Uptane::Role delegate_role = Uptane::Role::Delegation(delegate_name);
+    auto patterns = cur_targets.paths_for_role_.find(delegate_role);
+
+    if (patterns == cur_targets.paths_for_role_.end()) {
+      continue;
+    }
+
+    bool match = false;
+
+    for (const auto &pattern : patterns->second) {
+      if (fnmatch(pattern.c_str(), queried_target.filename().c_str(), 0) == 0) {
+        match = true;
+        break;
+      }
+    }
+
+    if (!match) {
+      continue;
+    }
+
+    // Target name matches one of the patterns
+
+    std::string delegation_meta;
+    std::shared_ptr<Uptane::Targets> delegation;
+
+    if (storage->loadDelegation(&delegation_meta, delegate_role)) {
+      delegation = Uptane::ImagesRepository::verifyDelegation(delegation_meta, delegate_role, cur_targets);
+      if (delegation == nullptr) {
+        storage->deleteDelegation(delegate_role);
+        delegation_meta.clear();
```

Comment:
Just to make sure I have this straight: we only delete a delegation if verification fails? So if a delegation is removed and later re-added, it wouldn't get deleted after removal, but hopefully verification will fail and then it will get deleted. Right?